### PR TITLE
Phase II.H: Shared DA Minters (w/o settlement)

### PR DESCRIPTION
**Pull request by ryley-o**
## Description of the change

Add handlers for new shared DA minters without settlement.

These updates include handling all events from `ISharedMinterDAV0`, `ISharedMinterDAExpV0`, and `ISharedMinterDALinV0`.

Tests

- e2e tests are added for all handled events
- (stated for clarity) e2e tests are not added for events in other interfaces already tested, even if they are emitted by the new minters

---
Asana task: https://app.asana.com/0/0/1205078004803556/f

### CODE REVIEW

**Comment by ryley-o** [comment]
Cc @lyaunzbe

File: src/helpers.ts [hunk 1]
```diff
@@ -382,24 +382,44 @@ export function loadOrCreateMinter(
...
-        contract.minimumAuctionLengthSeconds(),
-        minter
-      );
+      // only do this for legacy minters, because the new minters emit events
+      if (
+        minter.type.endsWith("V0") ||
+        minter.type.endsWith("V1") ||
+        minter.type.endsWith("V2") ||
+        minter.type.endsWith("V3") ||
+        minter.type.endsWith("V4")
```
Comment (jakerockland) [inline src/helpers.ts:391]: Nit: can probably move this check into a lil util method within this file, given that we use it in a couple places.

File: src/helpers.ts [hunk 2]
```diff
@@ -478,3 +498,41 @@ export function updateProjectIfMinterConfigIsActive(
...
+ * Calculates the total auction time for an exponential Dutch auction.
+ * @param startPrice The starting price of the auction.
+ * @param basePrice The base price of the auction.
+ * @param halfLifeSeconds The half life of the auction.
+ * @returns The total auction time in seconds.
+ **/
+export function getTotalDAExpAuctionTime(
+  startPrice: BigInt,
+  basePrice: BigInt,
+  halfLifeSeconds: BigInt
```
Comment (jakerockland) [inline src/helpers.ts:512]: It seems like it is probably worthwhile to add some smol unit tests for this helper specifically?

**Review by jakerockland — APPROVED** [review]

File: src/helpers.ts (hunk 2 above)
Comment (ryley-o) [reply src/helpers.ts:512]: A single auction is checked via independently written typescript calculation in e2e tests, but agree here and added a couple of spot checks for this function against current prod hasura values in [55bb4ff](https://github.com/ArtBlocks/artblocks-subgraph/pull/241/commits/55bb4ff5fb09a496df4879ed4562b3302505ab31)

File: src/helpers.ts (hunk 1 above)
Comment (ryley-o) [reply src/helpers.ts]: offloaded to util functions in [1fa1ce1](https://github.com/ArtBlocks/artblocks-subgraph/pull/241/commits/1fa1ce1ceaaf4f54856c7a89942a3b0abaf68946)

**Comment by ryley-o** [comment]
Resolved all current feedback items, confirmed this is passing e2e tests, and will merge when @lyaunzbe has bandwidth to review 👍

**Review by lyaunzbe — APPROVED** [review]

**Pull request merged**